spyder/plugins/projects/plugin.py

```diff
@@ -188,18 +189,22 @@ def on_editor_available(self):
         treewidget.sig_renamed.connect(self.editor.renamed)
         treewidget.sig_tree_renamed.connect(self.editor.renamed_tree)
         treewidget.sig_module_created.connect(self.editor.new)
-        treewidget.sig_file_created.connect(
-            lambda t: self.editor.new(text=t))
 
-        self.sig_project_loaded.connect(
-            lambda v: self.editor.setup_open_files())
-        self.sig_project_closed[bool].connect(
-            lambda v: self.editor.setup_open_files())
+        self._editor_new = lambda t: self.editor.new(text=t)

```

Comment:
Let's create a proper method for this, so it's easier to reference it later. I mean, instead of adding it inline here, please add it below like this

```python
@Slot(string)
def _new_editor(self, text):
    self.editor.new(text=text)
```